### DISCUSSION TITLE
Genre Filter

### DISCUSSION THREAD

**Discussion by SoongNoonien** [SoongNoonien/mpdevil#43] [Q&A] [answered]
Hi!
I'm wondering if anyone is using the genre filter. I'd like to get rid of the large genre button in the artist list and make a smaller one in the header bar (at least in CSD-mode). I'm currently thinking of an extra side pane which can be toggled via a small button. Here is a preview of this without the button:
![Bildschirmfoto_2021-09-27_22-12-39](https://user-images.githubusercontent.com/49582808/134978918-765249bd-121c-4de4-a7bf-585f75aee017.png)

So, if anyone is using the genre filter please let me know how you use it and if the extra sidebar would be an improvement. If you are not using it please let me know too so I can see if it is needed at all (and if anyone is reading these "Discussions" ;-)).

**Comment by dgcampea** [comment, marked as answer]
I occasionally use it. The current layout "seems fine" but in theory the place where the genre button is could be repurposed for an "albumartist searchbox" too (if it's still on the plan).
I can see the ability to list by genre useful with a properly tagged music library though, I wouldn't completely remove it.

**Reply by SoongNoonien** [reply]
> I occasionally use it. The current layout "seems fine"...

So, do you think it is needed to activate/deactivate the genre filter with a dedicated toggle button in the GUI, or is a toggle in the preferences enough? Both could be easily combined with a keyboard shortcut. Or do you think the new layout is worse?

> ...the place where the genre button is could be repurposed for an "albumartist searchbox" too (if it's still on the plan).

Yes it is still on the plan. That's the reason why I started thinking about the genre filter again. When the artist list gets an optional permanent searchbox the genre list should get one too.

**Reply by dgcampea** [reply]
>  toggle in the preferences enough

A toggle in preferences sounds fine.

> Or do you think the new layout is worse?

IMO, it overloads the UI a bit. I seldomly use the 'genre' filter, having it always on feels like something you'd intensively use.
The space could be potentially used for other, more generic uses (not sure which ones but take a look at "Cantata" for instance).
Though of course, this is subjective.
PS: here with a poorly tagged library, it doesn't look pretty 😂

**Reply by SoongNoonien** [reply]
> IMO, it overloads the UI a bit. I seldomly use the 'genre' filter, having it always on feels like something you'd intensively use.

Would a keyboard shortcut or a small toggle button in the GUI fix this?

**Reply by dgcampea** [reply]
toggle sounds better

**Reply by SoongNoonien** [reply]
It's now in master. I added a toggle and ctrl+g as shortcut.
